### DISCUSSION TITLE
EntityFramework - gerando migrations

### DISCUSSION THREAD

**Discussion by marcelocaselli** [balta-io/forum#578] [Tira Dúvidas] [answered]
boa tarde, estou com o seguinte erro ao add migration: 
![erroMigration](https://github.com/balta-io/forum/assets/111652069/877a4be2-0070-4ae8-8769-e669048fcb98)

Pode ajudar?

abs

**Comment by marcelocaselli** [comment]
ja instalei o  Microsoft.EntityFrameworkCore.Design mesmo assim não foi

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @marcelocaselli, como vai?

Faltou um construtor no seu DbContext:
```csharp
public BlogDataContext(DbContextOptions options): base(options) { }
```

Adiciona este método entre a linha 9 e 10, testa e me retorna aqui.

**Comment by marcelocaselli** [comment]
![programcs](https://github.com/balta-io/forum/assets/111652069/76fea20e-ae3f-4318-86ef-79441f714dbd)

adicionei, está dando esse erro

**Reply by andrebaltieri** [reply]
Agora o erro é por que seu Program.cs está vazio... você apagou o conteúdo dele?

**Reply by marcelocaselli** [reply]
eu uso .net 8.....ele vem assim, pra gerar as migration qual conteudo deve conter o Program .cs?

**Reply by andrebaltieri** [reply]
@marcelocaselli o ideal é seguir igual o curso!
No caso, você pode copiar o conteúdo do Program.cs que apresento no repositório do curso para manter o padrão.

**Reply by marcelocaselli** [reply]
bom dia Balta, agora o código está igual. No Program cs esta apresentando esse erro antes de compilar, deve ser porque fizemos o construtor no DBContext, está pedindo um parâmentro. Seria o parâmetro a string de conexão?
![programcs2](https://github.com/balta-io/forum/assets/111652069/25a2e458-aeda-47be-917e-e5435e806aaf)

**Reply by marcelocaselli** [reply]
eu tirei o construtor e gerou a pasta migration, fiz o update e funcionou, devia ser alguma referência que estava faltando. 
Obrigado....seguimos em frente abraço

**Reply by andrebaltieri** [reply]
Show!! Se puder, coloca o exemplo final aqui, pode ajudar outras pessoas!
